### PR TITLE
Add tests to validate spatial aggregation

### PR DESCRIPTION
**Pull request by cijothomas**
Added tests to validate SDK is doing "spatial aggregation" correctly. That term is no longer used explicitly in the spec, but the expected behavior is not currently implemented. Based on my experience, many implementations had/have this bug.

eg: .NET had this bug from day1 and not fixed as of today. https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs#L883-L886

Related discussions on this topic:
https://github.com/open-telemetry/opentelemetry-specification/issues/2905
https://github.com/open-telemetry/opentelemetry-specification/issues/1874

The tests are ignored now (they won't pass today!), but I think its worth while to have them, while we work on Metrics SDK, to help us keep in mind some edge scenarios.

I'll dig more into fixing these. The shared discussions above only affected Observable instruments, but we are failing for Non-observable as well here, so there are likely more changes required.

### CODE REVIEW

File: opentelemetry-sdk/src/metrics/mod.rs [hunk 1]
```diff
@@ -262,4 +262,163 @@ mod tests {
...
+        let sum = metric
+            .data
+            .as_any()
+            .downcast_ref::<data::Sum<u64>>()
+            .expect("Sum aggregation expected for ObservableCounter instruments by default");
+
+        // Expecting 1 time-series only, as the view drops all attributes resulting
+        // in a single time-series.
+        // This is failing today, due to lack of support for spatial aggregation.
+        assert_eq!(sum.data_points.len(), 1);
```
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/mod.rs:346]: Also, adding a  test to validate the value of aggregation?

File: opentelemetry-sdk/src/metrics/mod.rs [hunk 2]
```diff
@@ -262,4 +262,163 @@ mod tests {
...
+        let sum = metric
+            .data
+            .as_any()
+            .downcast_ref::<data::Sum<u64>>()
+            .expect("Sum aggregation expected for Counter instruments by default");
+
+        // Expecting 1 time-series only, as the view drops all attributes resulting
+        // in a single time-series.
+        // This is failing today, due to lack of support for spatial aggregation.
+        assert_eq!(sum.data_points.len(), 1);
```
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/mod.rs:422]: same as above - Checking for the aggregated value?

**Review by lalitb — APPROVED** [review]
LGTM. Thanks. We have the same problem in otel-cpp for the Observable counter. I think synchronous works fine.

**Comment by cijothomas** [comment]
Merging. I'll look at fixing this for sync instruments.
 
Fixing for Observable is likely harder - haven't been able to solve in .NET, C++ yet, unfortunately :-( Hopefully Rust can fix that.)

**Pull request merged**